src/invocation/InvocationService.ts

```diff
@@ -412,6 +456,36 @@ export class InvocationService {
         });
     }
 
+    fetchSchema(schemaId: Long): Promise<void> {
+        return this.schemaService.fetchSchema(schemaId);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    registerSchema(schema: Schema, clazz: Function | undefined): Promise<void> {
+        return this.schemaService.put(schema).then(() => {
+            this.serializationService.registerSchemaToClass(schema, clazz);
+        })
+    }
+
+    private fetchSchemaAndTryAgain(schemaId: Long, clientMessage: ClientMessage, invocation: Invocation): void {
```

Comment:
I think this function should return a promise.

Reply:
Resolving since we discussed via Slack